docs/learn/learn-comparison-rollups.md

```diff
@@ -0,0 +1,155 @@
+included if no challenge is presented (and the required proofs are in place).
+
+While optimistic rollups provide scalability, they have both pros and cons to their approach:
+
+**Pros:**
+
+- They aren't limited by the type of state change - any state change can be included, meaning
+  existing apps do not have to account for it.
+- They can be parallelized for scalability.
+- A substantial amount of data can fit within a single rollup
+  ([in the case of Ethereum, for example](https://ethereum.org/en/developers/docs/scaling/optimistic-rollups/#scaling-ethereum-with-optimistic-rollups),
+  tens of thousands of transactions in a single state transition).
+
+**Cons:**
+
+- Transaction censorship and centralization are concerns - sequencers/layer two nodes can be
+  compromised in this regard.
+- Challenge periods could take a substantial amount of time to pass, increasing time for the rollup
+  to finalize onto the layer one network.
+- Due to their generalist nature of including any state change for their parent network, optimistic
+  rollups can run into gas limitations or cause network congestion in the case of Ethereum.
+
+Optimistic rollups are often used in the Ethereum ecosystem. Examples of optimistic EVM-based rollup
+solutions include:
+
+- [Optimisim](https://www.optimism.io/)
+- [Arbitrum](https://bridge.arbitrum.io/)
+- [Unipig](https://unipig.exchange/welcome)
+
+## Zero-knowledge Rollups
+
+Zero-knowledge rollups (often called ZK rollups) are a non-interactive method that utilizes
+zero-knowledge proofs to compute the validity of a particular set of state changes. Whereas
+optimistic rollups relied on fraud proofs, ZK rollups rely on cryptographic validation in the form
+of ZK proofs.
+
+Zero-knowledge rollups are significantly faster in finalization, as the cryptographic validity proof
+handles the nuance of ensuring a rollup is valid. However, the ZK rollups often suffer from
+performance due to their complexity and are hard to integrate into resource-constrained
+environments. Because Turing completeness is also difficult to achieve due to this computational
+overhead, their ability to be generalized (in terms of blockspace) is reduced. However, they have an
+extremely promising future in solving some of the problems of optimistic rollups and addressing
+secure scalability.
+
+**Pros:**
+
+- They don't require a large amount of data availability. Often, the proof is enough to ensure
+  validity.
+- Rollups able to be proven trustlessly.
+- Because the proof is immediately available, finality is also instantaneous.
+- They have a promising future overall, as they haven't reached their full potential yet.
+
+**Cons:**
+
+- Suffer from the same problems that other layer two solutions face regarding centralization of
+  layer two operators.
+- They are computationally expensive, and ZK circuits are difficult to implement.
+- The potential for congestion is still a factor, as the amount of data could still be problematic.
```

Comment:
```suggestion
Zero-knowledge rollups (often called ZK rollups) are a non-interactive method that utilizes
zero-knowledge proofs to compute the validity of a particular set of state changes. Whereas
optimistic rollups relied on fraud proofs, ZK rollups rely on cryptographic validation in the form
of ZK proofs.

Zero-knowledge rollups are significantly faster in finalization, as the cryptographic validity proof
handles the nuance of ensuring a rollup is valid. However, the ZK rollups often suffer from
performance due to their complexity and difficult implementation into resource-constrained
environments. Because Turing completeness is also challenging to achieve due to this computational
overhead, their ability to be generalized (in terms of blockspace) is reduced. However, they have a promising future in solving some of the problems of optimistic rollups and addressing
secure scalability.

**Pros:**

- They only require a small amount of data availability. Often, the proof is enough to ensure
  validity.
- They can be proven trustlessly.
- Because the proof is immediately available, finality is also instantaneous.
- They have a promising future overall, as they have not reached maturity yet.

**Cons:**

- They suffer from the same problems that other L2 solutions have regarding the centralization of
  L2 operators.
- They are computationally expensive, and ZK circuits are difficult to implement.
- The potential for congestion is still a factor, as the amount of data could still be problematic.
```